root/usr/lib64/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -76,16 +76,17 @@ def start_discovery_service
 end
 
 def configure_network static, mac, ip=nil, gw=nil, dns=nil, vlan=nil
-  command("systemctl stop foreman-proxy")
+  command("systemctl stop foreman-proxy", false)
   if static
     command("nm-configure primary-static '#{mac}' '#{ip}' '#{gw}' '#{dns}' '#{vlan}'")
   else
     command("nm-configure primary '#{mac}' '#{vlan}'")
   end
-  result = command("nmcli connection up primary", false)
-  command("nm-online -s -q --timeout=45") unless static
+  # NetworkManager monitors config files and auto connects immediately
+  wait = cmdline('fdi.nmwait', 120)
+  result = command("nm-online -s -q --timeout=#{wait}")
   # restarting proxy with regenerated SSL self-signed cert
-  command("systemctl start foreman-proxy") if result
```

Comment:
It always returns true. More than that, `nm-online` is actually *useless* tool, it does return even when network is not yet up. I had to workaround this above - via the wait loop.

After *hours and hours* of googling and trying, I am pretty confident to say - there is no way in systemd/NetworkManager to wait until network is initialized via DHCP. At least `nm-online` wont you get that. I think that's on purpose, they do not want services to be waiting for that. We need this tho, because we need the IP for SSL cert.